File: src/mb/mizinkobusters/kitpvp/listener/ContentsClickListener.java
package mb.mizinkobusters.kitpvp.listener;

import mb.mizinkobusters.kitpvp.utils.KitPvPUtils;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryType;
import org.bukkit.inventory.ItemStack;

public class ContentsClickListener implements Listener {

    @EventHandler(priority = EventPriority.LOWEST)
    public void onClick(InventoryClickEvent event) {
        Player player = (Player) event.getWhoClicked();
        if (!KitPvPUtils.isInWorld(player)) {
            return;
        }
        if (!KitPvPUtils.hasKit(player)) {
            event.setCancelled(true);
        }

        ItemStack item = event.getCurrentItem();
        if (item == null) {
            return;
        }

        if (event.getSlotType().equals(InventoryType.SlotType.ARMOR)) {
            event.setCancelled(true);
        }
    }
}

File: src/main/java/mb/mizinkobusters/kitpvp/listener/ContentsClickListener.kt
package mb.mizinkobusters.kitpvp.listener

import mb.mizinkobusters.kitpvp.utils.KitPvPUtils
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.inventory.InventoryType

class ContentsClickListener : Listener {
    @EventHandler(priority = EventPriority.LOWEST)
    fun onClick(event: InventoryClickEvent) {
        val player = event.whoClicked as Player
        if (!KitPvPUtils.isInWorld(player)) {
            return
        }
        if (!KitPvPUtils.hasKit(player)) {
            event.isCancelled = true
        }
        event.currentItem ?: return
        if (event.slotType == InventoryType.SlotType.ARMOR) {
            event.isCancelled = true
        }
    }
}